src/logic/sync/failed-profiles-retrier.ts

```diff
@@ -18,12 +18,28 @@ export function createFailedProfilesRetrierComponent({
 
       const failedDeployments: Sync.FailedProfileDbEntity[] = await db.getFailedProfileFetches(FIFTY_ITEMS)
 
-      if (failedDeployments.length === 0) {
+      const profilesToRetry: Sync.FailedProfileDbEntity[] = []
+      for (const failedDeployment of failedDeployments) {
+        const currentStoredProfile = await db.getProfileByPointer(failedDeployment.pointer)
+
+        if (currentStoredProfile && currentStoredProfile.timestamp > failedDeployment.timestamp) {
+          logger.info('A newer profile already exists in database, skipping retry', {
+            entityId: failedDeployment.entityId,
+            pointer: failedDeployment.pointer
+          })
+          await db.deleteFailedProfileFetch(failedDeployment.entityId)
+          continue
+        } else {
+          profilesToRetry.push(failedDeployment)
+        }
+      }
+
+      if (profilesToRetry.length === 0) {
```

Comment:
WDYT about solving the deletion in the query directly? I imagine something like:
```js
const failedDeployments: Sync.FailedProfileDbEntity[] = await db.getFailedProfileFetches(FIFTY_ITEMS)
const deletedFailedDeployments = await db.deleteFailedProfileFetches(failedDeployments)
const profilesToRetry = failedDeployments.filter(fd => !deletedFailedDeployments.find(dfd => fd === dfd)) // passing a proper callback to the filter function
```

Reply:
hey good one :)

I left it like:
```
      await db.removeDanglingFailedProfileFetches()
      const failedDeployments: Sync.FailedProfileDbEntity[] = await db.getFailedProfileFetches(FIFTY_ITEMS)
```